### PR TITLE
Testmodulestakenpriortoothers

### PR DESCRIPTION
**Pull request by a0129998**
created  testcases for modules taken prior to others

things to check for
1. testcases covers all major cases

PS, I pylinted the file.

### CODE REVIEW

File: test/test_modules_taken_before_others.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+'''
+    test_modules_before_others_ui.py tests the page views for
```
Comment (nlzz22) [inline test/test_modules_taken_before_others.py]: docstring file name is different from the actual file name. 
The docstring name is "test_modules_before_others_ui.py"
while the real name is "test_modules_taken_before_others.py"

File: test/test_modules_taken_before_others.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+
+from paste.fixture import TestApp
+from nose.tools import assert_equal, raises
+from app import APP
+from components import session
+
+class TestCode(object):
+    '''
+        tests the ui elemetents of modules taken before another module
```
Comment (nlzz22) [inline test/test_modules_taken_before_others.py]: "elemetents" should be spelled as "elements"

File: test/test_modules_taken_before_others.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        root.mustcontain(self.CONTENT_FORM_TITLE)
+        root.mustcontain(self.CONTENT_FORM_MODULEA)
+        root.mustcontain(self.CONTENT_FORM_MODULEB)
+        root.mustcontain(self.CONTENT_FORM_SELECT)
+        root.mustcontain(self.CONTENT_TABLE)
+        root.mustcontain(self.CONTENT_BUTTON)
+
+    def test_modules_taken_before_others_submit(self):
+        '''
+            tests taht the submit button will cause a redirect
```
Comment (nlzz22) [inline test/test_modules_taken_before_others.py]: "taht" is misspelled, should be "that"

File: test/test_modules_taken_before_others.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+        form = root.forms__get()[self.FORM_NAME]
+        form.__setitem__("moduleA", "CS1010")
+        form.__setitem__("moduleB", "CS2010")
+        form.__setitem__("aysem", "AY 16/17 Sem 2")
+
+        response = form.submit()
+        #check response is 200 OK
+        assert_equal(response.status, 200)
+        response.mustcontain()
```
Comment (nlzz22) [inline test/test_modules_taken_before_others.py]: Why is there nothing passed into the mustcontain(), then what is this line checking for ?

**Comment by a0129998** [comment]
@nlzz22 updated changes

**Pull request merged**